### PR TITLE
Replace creating dag with dbt graph builder

### PR DESCRIPTION
**Pull request by ptutak-getindata**
Replace creating dag with dbt-graph-builder library

Resolves DATA-627

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.9]
+        python-version: ["3.10"]
```
Comment (p-pekala) [inline .github/workflows/publish.yml:13]: Will it work if someone use 3.9 to run this tool?

File: dbt_airflow_factory/tasks_builder/builder.py [hunk 2]
```diff
@@ -162,24 +164,16 @@ def _create_tasks_from_graph(self, dbt_airflow_graph: DbtAirflowGraph) -> ModelE
 
     def _make_dbt_tasks(self, manifest_path: str) -> ModelExecutionTasks:
         manifest = self._load_dbt_manifest(manifest_path)
-        dbt_airflow_graph = self._create_tasks_graph(manifest)
+        dbt_airflow_graph: DbtManifestGraph = create_tasks_graph(
+            self.gateway_config,
+            manifest,
+            self.airflow_config.enable_dags_dependencies,
+            self.airflow_config.show_ephemeral_models,
```
Comment (p-pekala) [inline dbt_airflow_factory/tasks_builder/builder.py:171]: Wouldn't be better to wrap the configuration into a config class? Open-close principle - for libs very important.

**Review by p-pekala — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (ptutak-getindata) [reply .github/workflows/publish.yml:13]: Unfortunately not.

File: dbt_airflow_factory/tasks_builder/builder.py (hunk 2 above)
Comment (ptutak-getindata) [reply dbt_airflow_factory/tasks_builder/builder.py:171]: This probably needs to be changed in the lib in the first place. But we can do that.

Comment (p-pekala) [reply dbt_airflow_factory/tasks_builder/builder.py:171]: yes please

Comment (ptutak-getindata) [reply dbt_airflow_factory/tasks_builder/builder.py]: Done.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (ptutak-getindata) [reply .github/workflows/publish.yml]: Updated to 3.8.

**Review by p-pekala — APPROVED** [review]

**Pull request merged**